### PR TITLE
solution

### PR DESCRIPTION
**Pull request by olenitut**
[DEMO LINK](https://olenitut.github.io/react_list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -21,73 +21,97 @@ enum SortType {
...
-  { sortType, isReversed }: ReorderOptions,
+  sortType: SortType,
+  isReversed: boolean,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
+  // console.log(sortType, isReversed);
+
+  switch (sortType) {
+    case 1:
```
Comment (SanyaBratashchuk) [inline src/App.tsx:33]:
```suggestion
    case SortType.ALPHABET:
```

Comment (SanyaBratashchuk) [reply src/App.tsx:33]: Fix it everywhere

File: src/App.tsx [hunk 2]
```diff
@@ -21,73 +21,97 @@ enum SortType {
...
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(0);
```
Comment (SanyaBratashchuk) [inline src/App.tsx:52]:
```suggestion
  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
```

File: src/App.tsx [hunk 3]
```diff
@@ -21,73 +21,97 @@ enum SortType {
...
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortType !== 0 || isReversed !== false) && (
```
Comment (SanyaBratashchuk) [inline src/App.tsx:100]:
```suggestion
        {(sortType !== 0 || isReversed) && (
```

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]
Use ENUM in a proper way

File: src/App.tsx [hunk 4]
```diff
@@ -21,73 +21,96 @@ enum SortType {
...
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${sortType !== SortType.ALPHABET ? 'is-light' : ''}`}
```
Comment (OlEzhA0) [inline src/App.tsx:73]: you can use [this library](https://www.npmjs.com/package/classnames)

File: src/App.tsx [hunk 5]
```diff
@@ -21,73 +21,96 @@ enum SortType {
...
-
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+
+  const neededGoods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+
+  const sortHandler = (type: SortType) => {
+    setSortType(type);
+  };
```
Comment (OlEzhA0) [inline src/App.tsx:57]: add useCallback where needed

**Review by OlEzhA0 — APPROVED** [review]
Hey! I approve your work but check my comments and apply changes

**Pull request closed without merging**